grakn-test/src/test/java/ai/grakn/test/graql/query/InsertQueryTest.java

```diff
@@ -329,9 +328,20 @@ public void testIterateMatchInsertResults() {
 
         Map<String, Concept> result1 = results.next();
         assertEquals(ImmutableSet.of("x"), result1.keySet());
-        assertTrue(qb.match(var().isa("language").has("name", "123").has("name", "HELLO")).ask().execute());
-        assertFalse(qb.match(var().isa("language").has("name", "456").has("name", "HELLO")).ask().execute());
 
+        AskQuery query123 = qb.match(var().isa("language").has("name", "123").has("name", "HELLO")).ask();
+        AskQuery query456 = qb.match(var().isa("language").has("name", "456").has("name", "HELLO")).ask();
+
+        //Check if one of the matches have had the insert executed correctly
+        boolean oneExists;
+        if(query123.execute()){
+            oneExists = !query456.execute();
+        } else {
+            oneExists = query456.execute();
+        }
+        assertTrue("A match insert was not executed correctly for only one match", oneExists);
```

Comment:
Beautiful time to use the XOR operator:
`boolean oneExists = query123.execute() ^ query456.execute();`